### PR TITLE
bus/nes: Added support for several Shenzhen Jncota RPGs.

### PR DESCRIPTION
**Pull request by 0kmg**
Software list items promoted to working (nes.xml)
---------------------------------------
Jing Ke Xin Zhuan (Chi, Shenzhen Jncota)
Sheng Huo Lie Zhuan (Chi)
Xiang Shuai Chuan Qi (Chi, Shenzhen Jncota)
Zhan Guo Feng Yun (Chi)

### CODE REVIEW

File: hash/nes.xml [hunk 1]
```diff
@@ -56607,19 +56607,18 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
...
-<!-- Shenzen Jincota -->
+<!-- Shenzhen Jncota -->
 <!-- Some of these partially work with mapper 4, 74, 227 or 240, but we need to figure out the proper behavior! -->
 
-<!-- any relation with jkxz ?? -->
-	<software name="jingkxz" supported="no">
-		<description>Jing Ke Xin Zhuan (Chi, Shenzen Jincota)</description>
+<!-- This is a later re-release of the Supertone game -->
+	<software name="jingkxz" cloneof="jkxz">
+		<description>Jing Ke Xin Zhuan (Chi, Shenzhen Jncota)</description>
```
Comment (cuavas) [inline hash/nes.xml:56615]: This should be "Jīngkē Xīn Chuán"

File: hash/nes.xml [hunk 2]
```diff
@@ -56633,16 +56632,15 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 		</part>
 	</software>
 
-<!-- any relation with shenghlz ?? -->
-	<software name="shenghuo" supported="no">
+<!-- This is a later re-release of the C&E game -->
+	<software name="shenghuo" cloneof="shenghlz">
 		<description>Sheng Huo Lie Zhuan (Chi)</description>
```
Comment (cuavas) [inline hash/nes.xml:56637]: This should be "Shènghuǒ Lièzhuàn"

File: hash/nes.xml [hunk 3]
```diff
@@ -56659,7 +56657,7 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 	<software name="dierciji" supported="no">
 		<description>Di Er Ci Ji Qi Ren Da Zhan (Chi)</description>
```
Comment (cuavas) [inline hash/nes.xml:56658]: This should probably be "Dì Èr Cì Jīqìrén Dàzhàn" - it’s possible there should be even fewer spaces in there, but depending on how you divide "第二次" (roughly “for the second time”) into “words”.

File: hash/nes.xml [hunk 4]
```diff
@@ -56681,7 +56679,7 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 	<software name="fankfb" supported="no">
 		<description>Fan Kong Feng Bao (Chi)</description>
```
Comment (cuavas) [inline hash/nes.xml:56680]: This should be "Fǎnkǒng Fēngbào"

File: hash/nes.xml [hunk 5]
```diff
@@ -56703,7 +56701,7 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 	<software name="fengshnb" supported="no">
 		<description>Feng Shen Bang (Chi)</description>
```
Comment (cuavas) [inline hash/nes.xml:56702]: This should be "Fēngshénbǎng"

File: hash/nes.xml [hunk 6]
```diff
@@ -56723,9 +56721,9 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 	</software>
 
 	<software name="xianscq" supported="no">
-		<description>Xiang Shuai Chuan Qi (Chi, Shenzen Jincota)</description>
+		<description>Xiang Shuai Chuan Qi (Chi, Shenzhen Jncota)</description>
```
Comment (cuavas) [inline hash/nes.xml:56724]: This should be "Xiāngshuài Chuánqí"

File: hash/nes.xml [hunk 7]
```diff
@@ -56766,15 +56764,15 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 		</part>
 	</software>
 
-	<software name="zgfyun" supported="no">
+<!-- This seems to be a later re-release of the Supertone game with an altered title -->
+	<software name="zgfyun" cloneof="jkxz">
 		<description>Zhan Guo Feng Yun (Chi)</description>
```
Comment (cuavas) [inline hash/nes.xml:56769]: This should be "Zhànguó Fēngyún"

File: hash/nes.xml [hunk 8]
```diff
@@ -56788,15 +56786,15 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 		</part>
 	</software>
 
-	<software name="xiakecq" supported="no">
+<!-- This seems to be a later re-release of the C&E game with an altered title -->
+	<software name="xiakecq" cloneof="shenghlz">
 		<description>Xia Ke Chuan Qi (Chi)</description>
```
Comment (cuavas) [inline hash/nes.xml:56791]: This should be "Xiákè Chuánqí"

File: hash/nes.xml [hunk 9]
```diff
@@ -56813,7 +56811,7 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 	<software name="chuliuxx" supported="no">
 		<description>Chu Liu Xiang Xin Zhuan (Chi)</description>
```
Comment (cuavas) [inline hash/nes.xml:56812]: This should be "Chǔliúxiāng Xīn Chuán"

File: hash/nes.xml [hunk 10]
```diff
@@ -56835,7 +56833,7 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 	<software name="zhanssj" supported="no">
 		<description>Zhan Shen Shi Jie (Chi)</description>
```
Comment (cuavas) [inline hash/nes.xml:56834]: This should be "Zhànshén Shìjiè"

File: hash/nes.xml [hunk 11]
```diff
@@ -56857,7 +56855,7 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 	<software name="jianxqy" supported="no">
 		<description>Jian Xia Qing Yuan (Chi)</description>
```
Comment (cuavas) [inline hash/nes.xml:56856]: This should be "Jiànxiá Qíngyuán"

File: hash/nes.xml [hunk 12]
```diff
@@ -56879,7 +56877,7 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 	<software name="shenmoda" supported="no">
 		<description>Shen Mo Da Lu (Chi)</description>
```
Comment (cuavas) [inline hash/nes.xml:56878]: This should be "Shén Mó Dàlù"

File: hash/nes.xml [hunk 13]
```diff
@@ -56922,7 +56920,7 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 	<software name="hundunds" supported="no">
 		<description>Yong Zhe Hei An Shi Jie - Hun Dun De Shi Jie (Chi)</description>
```
Comment (cuavas) [inline hash/nes.xml:56921]: This should be "Yǒngzhě Hēi'àn Shìjiè - Hùndùn De Shìjiè"

File: hash/nes.xml [hunk 14]
```diff
@@ -56944,7 +56942,7 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 	<software name="kdgszs" supported="partial">
 		<description>Kou Dai Guai Shou - Zuan Shi Ban (KT-008 PCB)</description>
```
Comment (cuavas) [inline hash/nes.xml:56943]: This should be "Kǒudài Guàishòu - Zuànshí Bǎn"

File: hash/nes.xml [hunk 15]
```diff
@@ -56965,7 +56963,7 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 	<software name="kdgszz" cloneof="kdgszs" supported="partial">
 		<description>Kou Dai Guai Shou - Zhen Zhu Ban (KT-008 PCB, Alt)</description>
```
Comment (cuavas) [inline hash/nes.xml:56964]: This should be "Kǒudài Guàishòu - Zhēnzhū Bǎn"

File: hash/nes.xml [hunk 16]
```diff
@@ -56986,7 +56984,7 @@ preliminary proto for the PAL version, still running on NTSC systems) or the gfx
 	<software name="kdgsbj" cloneof="kdgszs" supported="partial">
 		<description>Kou Dai Guai Shou - Bai jin Ban (KT-008 PCB, Alt 2)</description>
```
Comment (cuavas) [inline hash/nes.xml:56985]: This should be "Kǒudài Guàishòu - Báijīn Bǎn"

**Review by cuavas — CHANGES_REQUESTED** [review]
We don’t transliterate Chinese with spaces between syllables.  Also, don’t use the non-standard abbreviation “Chi” – use “CN” for the country China or “ZH” for Chinese language.

File: hash/nes.xml (hunk 1 above)
Comment (0kmg) [reply hash/nes.xml:56615]: Jing Ke, the protagonist of the game, is a historical figure: https://en.wikipedia.org/wiki/Jing_Ke. His name is written with a space. 传 is read zhuan in this context. The coinage "xinzhuan" appears in titles though I don't think the spacing is standardized. I've chosen maybe the more common way. A random example: https://en.wikipedia.org/wiki/Wong_Fei_Hung_Series.

File: hash/nes.xml (hunk 3 above)
Comment (0kmg) [reply hash/nes.xml:56658]: The title evokes WWII just as "World War Z" does for English speakers. I've chosen one of several romanization styles. See: https://zh.wiktionary.org/wiki/%E7%AC%AC%E4%BA%8C%E6%AC%A1%E4%B8%96%E7%95%8C%E5%A4%A7%E6%88%B0

File: hash/nes.xml (hunk 5 above)
Comment (0kmg) [reply hash/nes.xml:56702]: This is a historical novel. The spacing here seems to be standard. Reference: https://en.wikipedia.org/wiki/Fengshen_Yanyi

File: hash/nes.xml (hunk 9 above)
Comment (0kmg) [reply hash/nes.xml:56812]: Chu Liuxiang is a (fictional) person's name: https://en.wikipedia.org/wiki/Chu_Liuxiang. See earlier comment about "xinzhuan".

File: hash/nes.xml (hunk 12 above)
Comment (0kmg) [reply hash/nes.xml:56878]: Shenmo is a set phrase. It is commonly not spaced. Example: https://sinolit.wordpress.com/2017/10/23/what-on-earth-are-shenmo-xiaoshuo

**Comment by 0kmg** [comment]
With all due respect, the requested changes have almost nothing to do with this PR (I didn't write the preexisting software list materials and I'm trying to fix a company name typo here). Now, I don't mind making unrelated requested changes in most situations, but going forward I have NO interest in getting bogged down in the quagmire of romanization.

That being said—since I feel like I've been forced to deal with the issue now—first, I see little to no marginal utility in replacing spaced-by-character romanization (which will continue to be used indefinitely for Chinese outside of academic or official documents) by pinyin. My opinion aside, you've proposed introducing errors here in otherwise correct transliterations. You need a literate Chinese speaker familiar with issues of romanization to make these changes. Instead you get a literate Japanese speaker (I have a degree in Japanese linguistics) who has to muddle through for an hour looking up and verifying information (and still not being sure about every case). See further comments.

**Comment by cuavas** [comment]
The trouble is, the overall quality of MAME’s software lists is terrible, and the NES software list is probably the worst.  They don’t get better by themselves, as we’ve seen by the near total lack of improvement in years.  As you’re obviously already aware, it’s not just the metadata that’s in terrible shape, but they’re often misidentified, bad dumps, or hooked up to the wrong device types.  It’s a great example of how badly things can go when you bulk-import from dubious sources

With this in mind, I’m faced with several options:
1. Admit software lists were a mistake and MAMEdev lacks the resources to maintain them, rip them out completely, and leave curation of software collections to other groups (e.g. No-Intro and redump).
2. Try to fix it myself.
3. Remove everything that looks even slightly suspect, and only allow stuff back in if it meets standards.
4. Enforce a policy of, “You touch it, you fix it.”

Now while option 1 would make certain team members very happy, it’s no longer workable.  We now have drivers that depend on software lists to load media properly, and it’s touted as a convenient way for users to try out software for unfamiliar systems.  For some more obscure systems, MAME software lists are the *only* documentation for known media dumps.  This isn’t going to happen.

I’m already spread thin enough on the ground that option 2 isn’t going to happen.

The problem with option 3 is that having broken stuff around sometimes encourages people to fix it.  We see that more with non-working machine drivers.  For example, see the recent contributions to the LaserWriter driver – that likely wouldn’t have happened if the non-working skeleton driver hadn’t been there.  Poor quality software lists theoretically give an opportunity for people to improve them.

That leaves option 4 as the only way to ensure things actually improve.  It’s the approach we take with code – developers are expected to clean up poor quality or deprecated code when touching a module.

From working in Shanghai, I can tell you the “spaced by character” style of transliteration is dead for Mandarin Chinese.  There are now two generations that have grown up with Hanyu Pinyin and take it for granted.  (As an aside, Hanyu Pinyin has also largely displaced wubi zixing as an input method in Mandarin Chinese-speaking areas that use simplified characters.  Improvements to context-sensitive conversion have probably helped drive this.)  The style used in this software list right now isn’t helpful to anyone – it seems to be Hanyu Pinyin without tone marks and spaced by character.  No-one actually writes Chinese like that, and we gain nothing by maintaining it.

But all that aside, after looking at the NES software list again, maybe the best solution really is to kill it with fire.

**Comment by 0kmg** [comment]
I appreciate you going into detail about your thought process here, and I understand the difficulties involved in wanting to improve the situation and yet having limited resources and no ideal options. On the other hand, I don't have as bleak a view of the NES software list (or any of them really). It's a work in progress and it will probably always be a bit of a mess outside of official releases. At least for the moment I'm around and trying to leave things in hopefully better shape than I've found them.

As for issues of romanization, I wasn't trying to make an argument for one style or the other. Half of the reason I don't want to get involved with it is arguments over such issues. To me it makes no difference if it's pinyin, Wade-Giles, or the informal spaced style. At risk of going in circles I was simply pointing out that for the way things are currently transliterated they are correct as far as I could see, and I am wary of introducing errors in making corrections (alt_title corrections on the other hand, no problem, since you just have to make sure the characters are correct and I don't need to know, say, that Jing Ke has a standard spacing in Western writings.)

**Pull request merged**